docs/content/operations/including-extensions.md

```diff
@@ -13,22 +13,137 @@ Druid extensions can be specified in the `common.runtime.properties`. There are
+│   ├── snappy-java-1.1.1.6.jar
+│   ├── zkclient-0.3.jar
+│   └── zookeeper-3.4.6.jar
+└── mysql-metadata-storage
+    ├── jdbi-2.32.jar
+    ├── mysql-connector-java-5.1.34.jar
+    └── mysql-metadata-storage-0.8.0-rc1.jar
+```
 
-The Druid processes will then only load up jars from the local repository and will not try to go out onto the internet to find the maven dependencies.
+As you can see, there are three sub-directories ```druid-examples```, ```druid-kafka-eight``` and ```mysql-metadata-storage``` under ```druid_extensions```, each sub-directory denotes an extension that Druid might load.
+
+3) Prepare hadoop dependencies directories under root hadoop dependency directory. The file structure for hadoop dependencies is almost same as normal Druid extensions described above, except that there is an extra layer of folder that specifies which version of hadoop it contains. 
+
+Example:
+
+Suppose you specify `druid.extensions.hadoopDependenciesDir=/usr/local/druid/hadoop_druid_dependencies`, and you want to prepare both `hadoop-client` 2.3.0 and 2.4.0,
+
+Then under ```hadoop_druid_dependencies```, it should look like this,
+
+```
+hadoop_druid_dependencies/
+└── hadoop-client
+    ├── 2.3.0
+    │   ├── activation-1.1.jar
+    │   ├── avro-1.7.4.jar
+    │   ├── commons-beanutils-1.7.0.jar
+    │   ├── commons-beanutils-core-1.8.0.jar
+    │   ├── commons-cli-1.2.jar
+    │   ├── commons-codec-1.4.jar
+    ..... lots of jars
+    └── 2.4.0
+        ├── activation-1.1.jar
+        ├── avro-1.7.4.jar
+        ├── commons-beanutils-1.7.0.jar
+        ├── commons-beanutils-core-1.8.0.jar
+        ├── commons-cli-1.2.jar
+        ├── commons-codec-1.4.jar
+    ..... lots of jars
+```
+
+As you can see, under ```hadoop-client```, there are two sub-directories, each denotes a version of ```hadoop-client```.
+
+4) Tell Druid which extensions to load.  Now you have prepared your extension directories, if you want Druid to load a specific list of extensions under root extension directory, you need to specify `druid.extensions.toLoad`. Using the example above, if you want Druid to load ```druid-kafka-eight``` and ```mysql-metadata-storage```, then you can specify `druid.extensions.toLoad=["druid-kafka-eight", "mysql-metadata-storage"]`.
+
+If you specify `druid.extensions.toLoad=[]`, Druid won't load any extension from file system.
+
+If you don't specify `druid.extensions.toLoad`, Druid will load all the extensions under root extension directory.
+
+### How to prepare extensions directories using `tools pull-deps`?
+
+If you don't want to manually setup the extension directory described above, you can use `tools pull-deps` to generate it.
+
+1) Specify `druid.extensions.extensionsDir` and `druid.extensions.hadoopDependenciesDir` for `tools pull-deps`.  After `tools pull-deps` finishes, if you go to the localtions specified in `druid.extensions.extensionsDir` and `druid.extensions.hadoopDependenciesDir`, you will see the extension directories there.
```

Comment:
@fjy I created a new page for pull-deps with a full example and documentation, and referenced it in including-extensions page